+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+                            obj.w.N_vt(ii,start_fe:opts.N_finite_elements(ii)) = {{'N_vt', dims.n_tangents}, 0, inf, 1};
+                            obj.w.Alpha_vt(ii,start_fe:opts.N_finite_elements(ii)) = {{'Alpha_vt', dims.n_tangents}, 0, inf, 1};
+                        end
+                    end
+                end
+            end
+
+            % For c_n ~= 1 case
+            rbp = ~opts.right_boundary_point_explicit;
+            
+            % 3d vars: Variables defined on each rk stage
+            %          some of which are also defined at the initial point:
+            obj.w.x(0,0,opts.n_s) = {{['x_0'], dims.n_x}, model.x0, model.x0, model.x0};
+            obj.w.z(0,0,opts.n_s) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+            if model.friction_exists
+                switch opts.friction_model
+                  case 'Polyhedral'
+                    obj.w.gamma_d(0,0,opts.n_s) = {{'gamma_d', dims.n_c}, 0, inf, 0};
+                    obj.w.delta_d(0,0,opts.n_s) = {{'delta_d', dims.n_tangents}, 0, inf, 0};
+                  case 'Conic'
+                    obj.w.gamma(0,0,opts.n_s) = {{'gamma', dims.n_tangents}, 0, inf, 0};
+                    switch opts.conic_model_switch_handling
+                      case 'Plain'
+                        % nothing
+                      case {'Abs', 'Lp'}
+                        obj.w.p_vt(0,0,opts.n_s) = {{'p_vt', dims.n_tangents}, 0, inf, 0};
+                        obj.w.n_vt(0,0,opts.n_s) = {{'n_vt', dims.n_tangents}, 0, inf, 0};
+                    end
+                end
+            end            
+            for ii=1:opts.N_stages
+                if (opts.rk_representation == RKRepresentation.integral ||...
+                    opts.rk_representation == RKRepresentation.differential_lift_x)
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),0:(opts.n_s+rbp)) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                else
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),0) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),opts.n_s) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                end
+                if (opts.rk_representation == RKRepresentation.differential ||...
+                    opts.rk_representation == RKRepresentation.differential_lift_x)
+                    obj.w.v(ii,1:opts.N_finite_elements(ii),1:opts.n_s) = {{'v', dims.n_x}};
+                end
+
+                if opts.lift_velocity_state
+                    obj.w.z_v(ii,1:opts.N_finite_elements(ii),1:opts.n_s) = {{'z_v', dims.n_x}};
+                end
+                
+                obj.w.z(ii,1:opts.N_finite_elements(ii),1:(opts.n_s+rbp)) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+                obj.w.lambda_normal(ii,1:opts.N_finite_elements(ii),1:(opts.n_s)) = {{'lambda_normal', dims.n_c}, 0, 50, 0};
```

Comment:
quite arbitrary upper bound on lambda normal (and a quite low value). 
Take a particle with a mass of 50kg and you have an infeasible problem.

Reply:
Correct. This is a vestige of my testing. will replace with max in options.